File: 21557 - Mukesh J/Week10/45. Jump Game II/45. Jump Game II.java
class Solution {
    public int jump(int[] nums) {
        if(nums.length==1)
            return 0;
        int l=0, r=nums[0], jumpcount=0, r_next;
        while(r<nums.length-1)
        {
            r_next=r;
            for(int i=l;i<=r;i++)
            {
                r_next= Math.max(r_next, i+nums[i]);
            }
            jumpcount++;
            l= r+1;
            r= r_next;
        }
        return jumpcount+1;
    }
}

File: 21551-naveen-siva-kumar/week10/45_Jump Game II.py
class Solution {
public:
    int jump(vector<int>& nums) {
        
        int ans=1;
        int l=0,r=nums[0];
        int farthest=0;
        if(nums.size()==1)
            return 0;
        
        while(r<nums.size()-1){
            farthest=0;
            for(int i=l;i<r+1;i++){
                farthest=max(farthest,i+nums[i]);
            }
            l=r+1;
            r=farthest;
            ans+=1;
        }
        return ans;
    }
};
